generate/generate.go

```diff
@@ -182,54 +183,52 @@ func Do(root string, workingDir string) Report {
 		}
 		return report
 	})
-}
 
-func validateGeneratedFiles(generated []fileInfo) error {
-	logger := log.With().
-		Str("action", "generate.validateGeneratedFiles()").
-		Logger()
-
-	logger.Trace().Msg("validating generated files.")
-
-	genset := map[string]fileInfo{}
-	for _, file := range generated {
-		if other, ok := genset[file.Name()]; ok && file.Condition() {
-			return errors.E(ErrConflictingConfig,
-				"configs from %q and %q generate a file with same name %q have "+
-					"`condition = true`",
-				file.Origin(),
-				other.Origin(),
-				file.Name(),
-			)
-		}
+	outdatedDirs, err := listGenFilesOutsideStacks(rootdir, rootdir)
```

Comment:
I'm thinking that we could use this later to do outdated code detection (not actual removal). Didn't fixed it on this PR because it seemed like the PR would be too big, so this one is focused only on actually removing outdated files.